### PR TITLE
snap,client: show "publisher" in `snap list` and expose in client API

### PR DESCRIPTION
**Pull request by mvo5**
In the client package we have a "Developer" entry in the Snap
struct. However this name is actually misleading as this is
set to the publisher of the snap. To decrease confusion this
PR adds a "Publisher" field that is also set to the publisher.

This means that with PR#5234 (`snap list --format=help`) we
can now show a meaningful field.

This PR also modifies the output of `snap list` to show
"publisher" in the header instead of "developer" to reflect
reality.

### CODE REVIEW

**Comment by zyga** [comment]
@mvo5 there's a spread failure as a result of this change:
```
+ expected='^core .* [0-9]{2}-[0-9.]+(~[a-z0-9]+)?(\+git[0-9]+\.[0-9a-f]+)? +[0-9]+ +edge +canonical +core *$'
+ MATCH '^core .* [0-9]{2}-[0-9.]+(~[a-z0-9]+)?(\+git[0-9]+\.[0-9a-f]+)? +[0-9]+ +edge +canonical +core *$'
+ snap list
error: pattern not found, got:
Name              Version                     Rev   Tracking  Publisher  Notes
core              16-2.33~rc1+git776.efd90f3  4812  edge      -          core
test-snapd-tools  1.0                         x1    -         -          -
```

File: client/packages.go [hunk 1]
```diff
@@ -32,15 +32,17 @@ import (
...
+	ID            string    `json:"id"`
+	Title         string    `json:"title,omitempty"`
+	Summary       string    `json:"summary"`
+	Description   string    `json:"description"`
+	DownloadSize  int64     `json:"download-size,omitempty"`
+	Icon          string    `json:"icon,omitempty"`
+	InstalledSize int64     `json:"installed-size,omitempty"`
+	InstallDate   time.Time `json:"install-date,omitempty"`
+	Name          string    `json:"name"`
+	Publisher     string    `json:"publisher" help:"The publisher of the snap"`
```
Comment (chipaca) [inline client/packages.go]: Hmm.
Could you add a
```go
type  Publisher struct {
 ID string `json:"id"`
 Username string `json:"username"`
 DisplayName string `json:"display-name,omitempty"`
}
```
and then have
```go
Publisher Publisher `json:"publisher"`
```
pretty please?

Comment (mvo5) [reply client/packages.go]: Yes, this means some more work in daemon as well. But certainly doable.

**Comment by pedronis** [comment]
closing this because about to propose a new PR expanding on this and following @chipaca's suggestion also because that direction makes sense for 'verified' developers' work

**Pull request closed without merging**